packages/core/data-transfer/src/engine/index.ts

```diff
@@ -191,18 +193,22 @@ class TransferEngine<
     const { throttle } = this.options;
     const { global: globalTransforms, [key]: stageTransforms } = this.options?.transforms ?? {};
 
-    let stream = new PassThrough({ objectMode: true });
+    let stream: PassThrough | Chain = new PassThrough({ objectMode: true });
```

Comment:
Do we need this? Won't `stream` always be an instance of a stream (even if we're piping it to a chain)?

Reply:
nope, stream.pipe(chain) returns a Chain

Reply:
We could probably do something everywhere we chain to convert it back to a PassThrough but it might just complicate things. We could also probably eliminate chain altogether and do some extra work in building the chains manually with PassThroughs. That's why I would like to clean this up, but I'm not sure I'll have time.